app/Http/Controllers/OfferController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Offer;
+use App\Status;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Validator;
+
+class OfferController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            "user_id" => "required",
+            "category_id" => "required",
+            "start_at" => "required|date",
+            "end_at" => "required|date|after:start_at",
+            "price_per_hour" => "required|numeric|min:1|max:5000",
+            "address" => "required|max:500",
+            "preferred_qualifications" => "max:500"
+        ]);
+
+        if ($validator->fails()) {
+            return response()->json(["success" => false, "errors" => $validator->errors()], 400);
+        }
+
+        $offer = [
+            "user_id" => $request->user_id,
+            "category_id" => $request->category_id,
+            "start_at" => $request->start_at,
+            "end_at" => $request->end_at,
+            "price_per_hour" => $request->price_per_hour,
+            "address" => $request->address,
+            "preferred_qualifications" => $request->preferred_qualifications,
+            "status_id" => $this->getCreateStatusId(),
```

Comment:
We can make an associative array has the our two category as Enum and we can easily access it as it will be a content array.